pkg/util/validation/limits.go

```diff
@@ -228,6 +250,20 @@ func (l *Limits) Validate(shardByAllLabels bool) error {
 		return errMaxGlobalSeriesPerUserValidation
 	}
 
+	// If query priority is enabled, do not allow duplicate priority values
+	if l.QueryPriority.Enabled {
+		queryPriority := l.QueryPriority
+		prioritySet := map[int64]struct{}{}
+		prioritySet[queryPriority.DefaultPriority] = struct{}{}
+		for _, priority := range queryPriority.Priorities {
+			if _, exists := prioritySet[priority.Priority]; exists {
+				return errDuplicateQueryPriorities
+			}
+
+			prioritySet[priority.Priority] = struct{}{}
+		}
+	}
+
```

Comment:
Unrelated to this code. I am wondering if we can move the regex compilation to `Limits.UnmarshalYAML` and `Limits. UnmarshalJSON`.

This is the time when we deserialize the yaml/json format to `Limits` struct. Maybe doing it here is easier than compiling regex at query time.

Reply:
Are those functions only called when the limits config have changed? My only concern is that we compiling regex redundantly.

Reply:
The main change is to move compilation to load time, not during query time. Doing it when loading the config is a separate process and won't impact query workload. And easier to manage state.

> Are those functions only called when the limits config have changed
If we keep the current logic. Every reload we need to recompilie. I don't know if it is a big issue because it doesn't impact query time and we reload runtime config at a configurable interval (10s by defualt) in a separate goroutine.

But I think you can still avoid recompilation if needed by caching the prev state in runtime manager. That's probably easier as it doesn't need to hold the lock (it is single thread).

Another idea is to have a global LRU cache to cache `regex string -> compiled regex`.

Reply:
Addressed in https://github.com/cortexproject/cortex/pull/5605/commits/22be0b8a827db870eff985ba56c1bd2807b3107a